# Cargar_datos.ipynb

## Propósito

Este notebook tiene un único objetivo: **cargar el dataset histórico de créditos** (`Base_de_datos.xlsx`) en memoria como un DataFrame de pandas, dejándolo disponible para los análisis posteriores.

No se realiza aquí ningún análisis exploratorio, transformación ni limpieza. El análisis exploratorio profundo (univariable, bivariable y multivariable) se desarrolla íntegramente en el notebook `comprension_eda.ipynb`, el cual carga el dataset de forma independiente.

## Contexto del proyecto

Proyecto Integrador del Módulo 5 - Soy Henry Data Science.

Se busca desarrollar un modelo predictivo supervisado para anticipar el comportamiento crediticio de nuevos usuarios (variable objetivo: `Pago_atiempo`).

## Versión

**V1.0.1** - Ingesta del dataset.

## 1. Importaciones

In [1]:
# Importación de la librería principal para manipulación de datos tabulares.
# pandas es el estándar de facto en el ecosistema Python para trabajar con DataFrames.
import pandas as pd

# Se importa también openpyxl para asegurar que pandas pueda leer archivos .xlsx.
# pandas usa openpyxl internamente como motor (engine) cuando lee Excel modernos.
# La importación explícita no es estrictamente necesaria, pero deja constancia
# de la dependencia del proyecto.
import openpyxl  # noqa: F401  (importado para documentar dependencia)

# Verificación rápida de versiones instaladas.
print(f'pandas:   {pd.__version__}')
print(f'openpyxl: {openpyxl.__version__}')

pandas:   2.3.3
openpyxl: 3.1.5


## 2. Definición de la ruta del dataset

El dataset `Base_de_datos.xlsx` se encuentra en la **raíz del repositorio**, según la estructura definida por el Proyecto Integrador.

Este notebook vive en `mlops_pipeline/src/`, por lo que la ruta relativa al dataset es subir dos niveles (`../../`) y luego acceder al archivo.

In [3]:
# Ruta relativa al dataset desde la ubicación del notebook.
# Estructura del repositorio:
#   repo/
#     Base_de_datos.xlsx        <-- archivo a cargar
#     mlops_pipeline/
#       src/
#         Cargar_datos.ipynb    <-- ubicación de este notebook
RUTA_DATASET = '../../Base_de_datos.xlsx'

print(f'Ruta configurada para el dataset: {RUTA_DATASET}')

Ruta configurada para el dataset: ../../Base_de_datos.xlsx


## 3. Carga del dataset

Se utiliza `pd.read_excel()` para leer el archivo. Como el archivo solo contiene una hoja (`Hoja1`), no es necesario especificar el parámetro `sheet_name`; pandas tomará la primera por defecto.

**Nota técnica:** la lectura de archivos Excel es notablemente más lenta que la de archivos CSV. Para datasets muy grandes en producción se recomienda convertir a CSV o formatos binarios optimizados (Parquet, Feather). Para este proyecto académico, el volumen (~10K filas) hace que el formato Excel sea aceptable.

In [4]:
# Carga del dataset desde el archivo Excel.
# El método read_excel devuelve directamente un DataFrame.
df = pd.read_excel(RUTA_DATASET)

# Confirmación de carga exitosa mediante las dimensiones del DataFrame resultante.
print(f'Dataset cargado correctamente.')
print(f'Dimensiones (filas, columnas): {df.shape}')

Dataset cargado correctamente.
Dimensiones (filas, columnas): (10763, 23)


## 4. Verificación mínima de la carga

Se realiza una única verificación mínima para confirmar que la carga fue exitosa: mostrar las primeras filas del DataFrame.

Esta es una **verificación de sanidad**, no un análisis exploratorio. El análisis exploratorio (tipos de datos, nulos, distribuciones, etc.) se realiza en su totalidad en `comprension_eda.ipynb`.

In [5]:
# Visualización de las primeras 5 filas como verificación de carga.
df.head()

,tipo_credito,fecha_prestamo,capital_prestado,plazo_meses,edad_cliente,tipo_laboral,salario_cliente,total_otros_prestamos,cuota_pactada,puntaje,...,saldo_mora,saldo_total,saldo_principal,saldo_mora_codeudor,creditos_sectorFinanciero,creditos_sectorCooperativo,creditos_sectorReal,promedio_ingresos_datacredito,tendencia_ingresos,Pago_atiempo
0,7,2024-12-21 11:31:35,3692160.0,10,42,Independiente,8000000,2500000,341296,88.768094,...,0.0,51258.0,51258.0,0.0,5,0,0,908526.0,Estable,1
1,4,2025-04-22 09:47:35,840000.0,6,60,Empleado,3000000,2000000,124876,95.227787,...,0.0,8673.0,8673.0,0.0,0,0,2,939017.0,Creciente,1
2,9,2026-01-08 12:22:40,5974028.4,10,36,Independiente,4036000,829000,529554,47.613894,...,0.0,18702.0,18702.0,0.0,3,0,0,NaN,NaN,0
3,4,2025-08-04 12:04:10,1671240.0,6,48,Empleado,1524547,498000,252420,95.227787,...,0.0,15782.0,15782.0,0.0,3,0,0,1536193.0,Creciente,1
4,9,2025-04-26 11:24:26,2781636.0,11,44,Empleado,5000000,4000000,217037,95.227787,...,0.0,204804.0,204804.0,0.0,3,0,1,933473.0,Creciente,1


In [6]:
# Listado de las columnas presentes en el dataset.
# Permite confirmar que las 23 columnas esperadas están presentes.
print(f'Número de columnas: {len(df.columns)}')
print('\nNombres de las columnas:')
for i, col in enumerate(df.columns, start=1):
    print(f'  {i:2d}. {col}')

Número de columnas: 23

Nombres de las columnas:
   1. tipo_credito
   2. fecha_prestamo
   3. capital_prestado
   4. plazo_meses
   5. edad_cliente
   6. tipo_laboral
   7. salario_cliente
   8. total_otros_prestamos
   9. cuota_pactada
  10. puntaje
  11. puntaje_datacredito
  12. cant_creditosvigentes
  13. huella_consulta
  14. saldo_mora
  15. saldo_total
  16. saldo_principal
  17. saldo_mora_codeudor
  18. creditos_sectorFinanciero
  19. creditos_sectorCooperativo
  20. creditos_sectorReal
  21. promedio_ingresos_datacredito
  22. tendencia_ingresos
  23. Pago_atiempo


## 5. Resumen

El dataset `Base_de_datos.xlsx` ha sido cargado correctamente en el DataFrame `df`.

- **Filas cargadas:** 10,763
- **Columnas cargadas:** 23
- **Variable objetivo:** `Pago_atiempo`
- **Formato fuente:** Excel (.xlsx)

El análisis profundo del contenido del dataset continúa en `comprension_eda.ipynb`.